yarn2-plugins/prune-dev-dependencies/src/index.ts

```diff
@@ -0,0 +1,41 @@
+import { Cache, Configuration, Plugin, Project, StreamReport, CommandContext } from '@yarnpkg/core'
+import { Command } from 'clipanion'
+
+class HerokuPruneDevDependenciesCommand extends Command<CommandContext> {
+  @Command.Path('heroku', 'prune')
```

Comment:
This syntax got deprecated in Yarn 3 and is removed in Yarn 4 (currently in rc); you should instead write:

```ts
static paths = [[`heroku`, `prune`]];
```

Reply:
Btw, I don't know if you're aware, but you can prune dev dependencies with [`yarn workspaces focus`](https://yarnpkg.com/cli/workspaces/focus)

Reply:
Thanks for the feedback @arcanis, while the `yarn workspaces focus --all --production` command does work as advertised to prune installed dependencies, we also want to prune these dependencies from the Yarn cache to further reduce slug sizes for compiled application.  Based on the discussions from https://github.com/yarnpkg/berry/issues/1789 and https://github.com/yarnpkg/berry/pull/1798 it doesn't sound that is a use case that you are willing to support.

Also, many of the tests we had did not work with the Yarn 3 plugin syntax.  This may have been a mistake on my part but the failures went away when I downgraded to the Yarn 2 plugin syntax 🤷 .  Either way, the custom plugin we're using here is operating on a limited percentage of builds currently but, if it does work well for our needs, we plan to update it with better coverage for Yarn 2, 3, and 4.